### DISCUSSION TITLE
Add botan backend to curl.

### DISCUSSION THREAD

**Discussion by talregev** [randombit/botan#4320] [Ideas]
Curl have a few backend list here:
https://curl.se/libcurl/c/tls-options.html
Very known and popular library that use in more 2 billion devices.
In their ci, there is a lot of testing, so adding a botan to curl, get helped test and use.
Add botan as an http3 backend option it will be wonderful.

There already old issue that I tried to revive. I also open this discussion:
https://github.com/randombit/botan/issues/1323 

Let me know what you think and if you want this as well, please vote / comment.

**Comment by reneme** [comment]
This has certainly crossed my mind and I think it makes a lot of sense. As far as I can tell from a quick skim of curl's TLS abstraction, we'd have to expose TLS via the Foreign Function Interface. This is listed in the library's TODOs literally for a decade now. 😲

Exposing the fundamental TLS functionality is a good chunk of work, but should be fairly straight forward. Picking curl as the first consumer for such a binding sounds like a sensible choice. Given the number of supported TLS backends in curl, I would assume that their TLS abstractions are in very good shape. Designing Botan's FFI binding to work well with curl's abstractions would therefore yield a decent API, I bet.

I'd be happy to support such a project.

**Comment by talregev** [comment]
If it still interesting you to do, curl accept new tls backends. here one that discuss about it: 
https://github.com/curl/curl/discussions/18429
